static/js/reactions.js

```diff
@@ -0,0 +1,194 @@
+var reactions = (function () {
+var exports = {};
+
+var reactions_store = new Dict();
```

Comment:
Why do you need this object?  It seems like you could just keep the reaction data in the relevant message_store object for the message; is there an advantage to having this extra dict off to the side?  It seems like it adds some extra accounting complexity...